examples/CMakeLists.txt

```diff
@@ -44,3 +44,23 @@
 raja_add_executable(
   NAME example-raja-pi
   SOURCES pi.cpp)
+
+raja_add_executable(
```

Comment:
The only thing I would change at this point is having unified naming for all of the example binaries that get emitted.

* `example-raja-pi` --> `example-pi`
* `addVectors` --> `example-vector-add`
* `matrixMultiply` --> `example-matrix-multiply`
* `jacobi` --> `example-jacobi`
* `wave` --> `example-wave`
* `customIndex` --> `example-custom-index`

Other than that, the source file names are appropriate for the examples :)